### PR TITLE
fix(api): Fix Access project/team membership checks for global roles

### PR DESCRIPTION
**Pull request by macqueen**
per a discussion in slack this:
- fixes a bug where for owners/managers, all projects are returned from `get_projects`
- renames a `open_access_policy` kwarg for `Access` to `has_global_access` so that it can be used more broadly without confusion

### CODE REVIEW

File: src/sentry/auth/access.py [hunk 1]
```diff
@@ -259,7 +267,8 @@ def from_request(request, organization=None, scopes=None):
             projects=project_list,
             sso_is_valid=sso_is_valid,
             requires_sso=requires_sso,
-            open_access_policy=True,
+            open_access_policy=bool(organization.flags.allow_joinleave),
```
Comment (dcramer) [inline src/sentry/auth/access.py]: maybe if we're going to add another flag we should just rename the existing flag to something like "has_global_access" and then I think it'd be more appropriate to combine the flags

my concern with yet-another flag is just theres a lot of branches things can go down to make a decision, and my concer w/ overloading "open_access_policy" is it becomes less obvious what it means

Comment (macqueen) [reply src/sentry/auth/access.py]: 👍 i like the idea of just combining into a more generic flag, will do that

**Review by dcramer — APPROVED** [review]
lgtm assuming tests pass

**Review by markstory — APPROVED** [review]
Looks good to me. The modified `get_teams()` method has no other callers that could be impacted.

**Pull request merged**